### PR TITLE
Add support and documentation for export from HDMF version 2

### PR DESCRIPTION
**Pull request by rly**
## Motivation

See https://github.com/hdmf-dev/hdmf/pull/388

I also changed the docval for `NWBHDF5IO.export` from what it was for `HDF5IO.export`:
```python
HDF5IO.export(path=..., container=...)
NWBHDF5IO.export(path=..., nwbfile=...)
```

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number?

### CODE REVIEW

File: docs/gallery/general/modifying_data.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
+"""
+.. _modifying_data:
+
+Modifying data in an NWB file
```
Comment (oruebel) [inline docs/gallery/general/modifying_data.py]: I think the term "modifying" is confusing here. The reason being that folks will typically think about modifying individual fields, i.e., datasets and attributes. I think we should either add a section on how to update datasets and attributes or change the title to something like "Adding/Removing containers from an NWB file".

File: docs/gallery/general/modifying_data.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+=========================================
+
+This tutorial explains how to modify data from an existing NWB file and either write the data back to the
```
Comment (oruebel) [inline docs/gallery/general/modifying_data.py]: Same here, we should explicitly state what we mean by "modify"

**Review by oruebel — DISMISSED** [review]
Aside from the minor comment the tutorial description this looks good.

**Comment by rly** [comment]
@oruebel I made the suggested changes and added another section on Object IDs

**Review by oruebel — APPROVED** [review]

**Pull request merged**